src/utils/actions.ts

```diff
@@ -16,8 +16,13 @@ export function onAcceptImprovement(
       state.selection.main.from,
       state.selection.main.to
     )
-    const changes = [];
-    const diffs = Diff.diffWordsWithSpace(originalContent, e.detail.improvement);
+    let changes = [];
+    let diffs = Diff.diffChars(originalContent, e.detail.improvement);
+
+    if (diffs.length >= 500) {
+      diffs = Diff.diffWords(originalContent, e.detail.improvement);
```

Comment:
It has to be `diffWordsWithSpace`. `diffWords` ignores some whitespaces and will cause issues when replacing.

Reply:
Ah, thanks for catching this! Should be fixed.